src/enhancers/convert_frequencies_to_stoptimes.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{
+    model::Collections,
+    objects::{Calendar, Frequency, StopTime, Time, VehicleJourney},
+    Result,
+};
+use anyhow::anyhow;
+use skip_error::skip_error_and_warn;
+use std::collections::{BTreeSet, HashMap};
+use tracing::warn;
+use typed_index_collection::CollectionWithId;
+
+/// Convert all frequencies to stoptimes
+/// by creating new duplicated vehicle_journeys, calendars and comments if necessary
+/// and remove all frequencies from Collections
+pub fn convert_all_frequencies_to_stoptimes(collections: &mut Collections) -> Result<()> {
+    let frequencies = collections.frequencies.take();
+    convert_frequencies_to_stoptimes(collections, frequencies)
+}
+
+pub fn convert_frequencies_to_stoptimes(
+    collections: &mut Collections,
+    frequencies: Vec<Frequency>,
+) -> Result<()> {
+    let mut trip_id_sequence: HashMap<String, u32> = HashMap::new();
+    let mut new_vehicle_journeys: Vec<VehicleJourney> = vec![];
+
+    for frequency in &frequencies {
+        if frequency.start_time == frequency.end_time {
+            warn!(
+                "frequency for trip {:?} has same start and end time",
+                frequency.vehicle_journey_id
+            );
+            continue;
+        }
+        let corresponding_vj = skip_error_and_warn!(collections
+            .vehicle_journeys
+            .get(&frequency.vehicle_journey_id)
+            .cloned()
+            .ok_or_else(|| anyhow!(
+                "frequency mapped to an unexisting trip {:?}",
+                frequency.vehicle_journey_id
+            )));
+        let mut start_time = frequency.start_time;
+        let mut arrival_time_delta = match corresponding_vj.stop_times.iter().min() {
+            None => {
+                warn!(
+                    "frequency mapped to trip {:?} with no stop_times",
+                    frequency.vehicle_journey_id
+                );
+                continue;
+            }
+            Some(st) => st.arrival_time,
+        };
```

Comment:
This might be a good case for the following syntax.

```rust
let Some(st) = corresponding_vj.stop_times.iter().min() else {
   warn!(
       "frequency mapped to trip {:?} with no stop_times",
       frequency.vehicle_journey_id
   );
   continue;
};
let mut arrival_time_delta = st.arrival_time;

Reply:
Done in https://github.com/hove-io/transit_model/pull/922/commits/99c1d2cfb85c68c9ce16fbeb3574aebed45e0175